src/app/public/modules/repeater/repeater.component.ts

```diff
@@ -33,36 +47,92 @@ export class SkyRepeaterComponent implements AfterContentInit {
   @ContentChildren(SkyRepeaterItemComponent)
   public items: QueryList<SkyRepeaterItemComponent>;
 
+  private ngUnsubscribe = new Subject<void>();
+
   private _expandMode = 'none';
 
-  constructor(private repeaterService: SkyRepeaterService) {
-    this.repeaterService.itemCollapseStateChange.subscribe((item: SkyRepeaterItemComponent) => {
-      if (this.expandMode === 'single' && item.isExpanded) {
-        this.items.forEach((otherItem) => {
-          if (otherItem !== item && otherItem.isExpanded) {
-            otherItem.isExpanded = false;
-          }
-        });
-      }
-    });
+  constructor(
+    private repeaterService: SkyRepeaterService
+  ) {
+    this.repeaterService.itemCollapseStateChange
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((item: SkyRepeaterItemComponent) => {
+        if (this.expandMode === 'single' && item.isExpanded) {
+          this.items.forEach((otherItem) => {
+            if (otherItem !== item && otherItem.isExpanded) {
+              otherItem.isExpanded = false;
+            }
+          });
+        }
+      });
 
     this.updateForExpandMode();
   }
 
   public ngAfterContentInit() {
+    // Initialize each item's index (in case items are instantiated out of order).
+    this.items.forEach(item => item.initializeItem());
+    this.items.changes
```

Comment:
This feels odd to me. We have a service which is used to manage the repeater items, but we're also managing the repeater items directly using `ContentChildren`. I would find a way to move this logic (and the other `this.items.changes`, below) to the service. 

In this specific case, shouldn't each repeater item's `ngOnInit` handle its own initialization with the service?

Reply:
Fair point. I've moved these registration steps to the repeater-item.component.